changelog/_unreleased/2024-07-22-add-cart-event-interface.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Abstract and reduce rule condition components
+issue: NEXT-00000
+author: Justus Maier
+author_email: jmaier@notebooksbilliger.de
+author_github: @justusNBB
+---
+# Core
+* Added `Shopware\Core\Checkout\Cart\Event\CartEvent interface` shared by all CartEvents supporting the getCart method
+* Added Cart- & SalesChannelEvent interfaces to CartEvents:
+  * `Core/Checkout/Cart/Event/AfterLineItemAddedEvent`
+  * `Core/Checkout/Cart/Event/AfterLineItemQuantityChangedEvent`
+  * `Core/Checkout/Cart/Event/AfterLineItemRemovedEvent`
+  * `Core/Checkout/Cart/Event/BeforeLineItemAddedEvent`
+  * `Core/Checkout/Cart/Event/BeforeLineItemQuantityChangedEvent`
+  * `Core/Checkout/Cart/Event/BeforeLineItemRemovedEvent`
+  * `Core/Checkout/Cart/Event/CartBeforeSerializationEvent`
+  * `Core/Checkout/Cart/Event/CartChangedEvent`
+  * `Core/Checkout/Cart/Event/CartContextHashEvent`
+  * `Core/Checkout/Cart/Event/CartCreatedEvent`
+  * `Core/Checkout/Cart/Event/CartEvent`
+  * `Core/Checkout/Cart/Event/CartLoadedEvent`
+  * `Core/Checkout/Cart/Event/CartMergedEvent`
+  * `Core/Checkout/Cart/Event/CartSavedEvent`
+  * `Core/Checkout/Cart/Event/CartVerifyPersistEvent`
+  * `Core/Checkout/Cart/Event/LineItemRemovedEvent`
+* Refactored `Shopware\Core\Checkout\Cart\Event\CartChangedEvent`: Rename method getContext to getSalesChannelContext for consistency
```

Comment:
```suggestion
* Changed `Shopware\Core\Checkout\Cart\Event\CartChangedEvent`: Rename method getContext to getSalesChannelContext for consistency
```

Reply:
Oh you are right, this is only a refactor internally, for consumers it's a change. I've also added lines to explain the deprecation.